realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -414,21 +413,31 @@ private boolean categorizeField(Element element) {
         return true;
     }
 
+    // The field has the @Index annotation. It's only valid for column types:
+    // STRING, DATE, INTEGER, BOOLEAN, and RealmMutableInteger
     private boolean categorizeIndexField(Element element, VariableElement variableElement) {
-        // The field has the @Index annotation. It's only valid for column types:
-        // STRING, DATE, INTEGER, BOOLEAN
-        Constants.RealmFieldType realmType = Constants.JAVA_TO_REALM_TYPES.get(variableElement.asType().toString());
-        if (realmType != null) {
-            switch (realmType) {
-                case STRING:
-                case DATE:
-                case INTEGER:
-                case BOOLEAN:
-                    indexedFields.add(variableElement);
-                    return true;
+        boolean indexable = false;
+
+        if (Utils.isMutableRealmInteger(variableElement)) {
```

Comment:
It would be nice, but that won't work.  It is treated differently than an Integer in a number of places, so putting it into the JAVA_TO_REALM_TYPES list won't work.